formatter.go

```diff
@@ -215,6 +237,19 @@ func (f *DefaultFormatter) buildFormatData(
 		}
 	}
 
+	var enableColor bool
+	switch f.ColorMode {
+	case ColorModeAuto:
+		fd := os.Stdout.Fd()
+		enableColor = testing.Verbose() &&
+			(isatty.IsTerminal(fd) || isatty.IsCygwinTerminal(fd))
```

Comment:
Ideally, this part of the check:

```
(isatty.IsTerminal(fd) || isatty.IsCygwinTerminal(fd))
```

should be done only once. We can achieve it by using sync.Once (it's better to use sync.Once instead of init() because this way we postpone the check until the first call to formatter and give the user a chance to manipulate with os.Stdout in test setup code).

Also, ColorModeAuto should disable colors by two additional conditions:

* if TERM env variable is set and starts with "dumb"
* or if NO_COLOR env variable is set and is non-empty